api-common/src/main/java/org/openmrs/module/metadatamapping/MetadataTermMapping.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.metadatamapping;
+
+import org.openmrs.BaseOpenmrsMetadata;
+import org.openmrs.OpenmrsMetadata;
+import org.openmrs.module.metadatamapping.util.ArgUtil;
+
+/**
+ * Maps a "term" withing a specific namespace or "source" to a local metadata object. {@link #getCode()} is the public 
+ * identifier of this term withing the {@link #getMetadataSource()}.
+ * @since 1.1
+ */
+public class MetadataTermMapping extends BaseOpenmrsMetadata {
+	
+	private Integer metadataTermMappingId;
+	
+	private MetadataSource metadataSource;
+	
+	private String code;
+	
+	private String metadataClass;
```

Comment:
In order to efficiently search for mapped items we need to create an index on metadataSource, code and class. The class name may be too long for a DB index (mysql innodb has 255 chars limit per column) so we will need an additional field to store a hashed (e.g. md5) version of metadataClass, which is short enough to be indexed. You can address that as a separate issue.